tests/test_attributes.py

```diff
@@ -1,86 +1,66 @@
-        self.assertEqual(
-            attributes_to_string({"class": "foo", "style": "color: red;"}),
-            'class="foo" style="color: red;"',
-        )
+        assert attributes_to_string({"class": "foo", "style": "color: red;"}) == 'class="foo" style="color: red;"'
 
     def test_escapes_special_characters(self):
-        self.assertEqual(
-            attributes_to_string({"x-on:click": "bar", "@click": "'baz'"}),
-            'x-on:click="bar" @click="&#x27;baz&#x27;"',
-        )
+        assert attributes_to_string({"x-on:click": "bar", "@click": "'baz'"}) == 'x-on:click="bar" @click="&#x27;baz&#x27;"'  # noqa: E501
 
     def test_does_not_escape_special_characters_if_safe_string(self):
-        self.assertEqual(
-            attributes_to_string({"foo": mark_safe("'bar'")}),
-            "foo=\"'bar'\"",
-        )
+        assert attributes_to_string({"foo": mark_safe("'bar'")}) == "foo=\"'bar'\""
 
     def test_result_is_safe_string(self):
         result = attributes_to_string({"foo": mark_safe("'bar'")})
-        self.assertTrue(isinstance(result, SafeString))
+        assert isinstance(result, SafeString)
 
     def test_attribute_with_no_value(self):
-        self.assertEqual(
-            attributes_to_string({"required": None}),
-            "",
-        )
+        assert attributes_to_string({"required": None}) == ""
 
     def test_attribute_with_false_value(self):
-        self.assertEqual(
-            attributes_to_string({"required": False}),
-            "",
-        )
+        assert attributes_to_string({"required": False}) == ""
 
     def test_attribute_with_true_value(self):
-        self.assertEqual(
-            attributes_to_string({"required": True}),
-            "required",
-        )
+        assert attributes_to_string({"required": True}) == "required"
 
 
-class AppendAttributesTest(BaseTestCase):
+@djc_test
+class TestAppendAttributes:
     def test_single_dict(self):
-        self.assertEqual(
-            append_attributes(("foo", "bar")),
-            {"foo": "bar"},
-        )
+        assert append_attributes(("foo", "bar")) == {"foo": "bar"}
 
     def test_appends_dicts(self):
-        self.assertEqual(
-            append_attributes(("class", "foo"), ("id", "bar"), ("class", "baz")),
-            {"class": "foo baz", "id": "bar"},
-        )
+        assert append_attributes(("class", "foo"), ("id", "bar"), ("class", "baz")) == {"class": "foo baz", "id": "bar"}  # noqa: E501
 
 
-class HtmlAttrsTests(BaseTestCase):
+@djc_test
+class TestHtmlAttrs:
     template_str: types.django_html = """
         {% load component_tags %}
         {% component "test" attrs:@click.stop="dispatch('click_event')" attrs:x-data="{hello: 'world'}" attrs:class=class_var %}
         {% endcomponent %}
     """  # noqa: E501
 
-    @parametrize_context_behavior(["django", "isolated"])
-    def test_tag_positional_args(self):
+    @djc_test(parametrize=PARAMETRIZE_CONTEXT_BEHAVIOR)
```

Comment:
The `djc_test` decorator also handles parametrization. This is available only when pytest is installed. In theory, the decorator should work also when pytest is not installed, as long as the `parametrize` kwarg is not used in the non-pytest environment.

And so I was also able replace and remove the `@parametrize_context_behavior` decorator. `@parametrize_context_behavior` was an attempt at having parametrized tests. But because Django / unittest didn't properly support that, then in the reports the parametrized tests actually showed up as one.

So one side effect is that then number of shown tests might up to double, as the parametrized tests now show up as separate tests in the report.